modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/dto/v1_0/converter/ProductOptionDTOConverter.java

```diff
@@ -57,17 +60,35 @@ public ProductOption toDTO(DTOConverterContext dtoConverterContext)
 				facetable = cpDefinitionOptionRel.isFacetable();
 				fieldType = cpDefinitionOptionRel.getDDMFormFieldTypeName();
 				id = cpDefinitionOptionRel.getCPDefinitionOptionRelId();
-				key = cpOption.getKey();
 				name = LanguageUtils.getLanguageIdMap(
 					cpDefinitionOptionRel.getNameMap());
-				optionId = cpOption.getCPOptionId();
 				required = cpDefinitionOptionRel.isRequired();
 				skuContributor = cpDefinitionOptionRel.isSkuContributor();
+
+				setKey(
+					() -> {
+						if (cpOption != null) {
+							return cpOption.getKey();
+						}
+
+						return StringPool.BLANK;
+					});
```

Comment:
```suggestion
					key = cpDefinitionOptionRel.getKey();
```


This was wrong in the beginning we should return the saved key for the CPOption, this is the most important field of the CPOption feature